scripts/clusterCreation/deployASB.sh

```diff
@@ -0,0 +1,609 @@
+
+  end_time=$(date +%s.%3N)
+  elapsed=$(echo "scale=3; $end_time - $start_time" | bc)
+  echo "Completed Deploying Default Hub. ($elapsed)"
+
+  export ASB_SCRIPT_STEP=deployDefaultSpoke
+  # Save environment variables
+  ./saveenv.sh -y
+
+  # Invoke Next Step In Setup
+  $ASB_SCRIPT_STEP
+}
+
+function deployDefaultSpoke()
+{
+  start_time=$(date +%s.%3N)
+
+  echo "Deploying Default Spoke..."
+
+  # Set spoke ip address prefix
+  export ASB_SPOKE_IP_PREFIX="10.240"
+
+  # Create spoke network
+  az deployment group create \
+    -n spoke-$ASB_ORG_APP_ID_NAME \
+    -g $ASB_RG_SPOKE \
+    -f networking/spoke-default.json \
+    -p deploymentName=${ASB_DEPLOYMENT_NAME} \
+      hubLocation=${ASB_HUB_LOCATION} \
+      hubVnetResourceId=${ASB_HUB_VNET_ID} \
+      orgAppId=${ASB_ORG_APP_ID_NAME} \
+      spokeIpPrefix=${ASB_SPOKE_IP_PREFIX} \
+      spokeLocation=${ASB_SPOKE_LOCATION} \
+    -c --query name
+
+  # Get nodepools subnet id from spoke
+  export ASB_NODEPOOLS_SUBNET_ID=$(az deployment group show -g $ASB_RG_SPOKE -n spoke-$ASB_ORG_APP_ID_NAME --query properties.outputs.nodepoolSubnetResourceIds.value -o tsv)
+
+  if [ -z $ASB_NODEPOOLS_SUBNET_ID ]; then echo "Step deployDefaultSpoke failed" 1>&2; exit 1; fi
+
+  end_time=$(date +%s.%3N)
+  elapsed=$(echo "scale=3; $end_time - $start_time" | bc)
+  echo "Completed Deploying Default Spoke. ($elapsed)"
+
+  export ASB_SCRIPT_STEP=deployHubRegionA
+  # Save environment variables
+  ./saveenv.sh -y
+
+  # Invoke Next Step In Setup
+  $ASB_SCRIPT_STEP
+}
+
+function deployHubRegionA()
+{
+  start_time=$(date +%s.%3N)
+
+  echo "Deploying Hub Region A..."
+
+  # Create Region A hub network
+  az deployment group create \
+    -g $ASB_RG_HUB \
+    -f networking/hub-regionA.json \
+    -p location=${ASB_HUB_LOCATION} nodepoolSubnetResourceIds="['${ASB_NODEPOOLS_SUBNET_ID}']" \
+    -c --query name
+
+  # Get spoke vnet id
+  export ASB_SPOKE_VNET_ID=$(az deployment group show -g $ASB_RG_SPOKE -n spoke-$ASB_ORG_APP_ID_NAME --query properties.outputs.clusterVnetResourceId.value -o tsv)
+
+  if [ -z $ASB_SPOKE_VNET_ID ]; then echo "Step deployHubRegionA failed" 1>&2; exit 1; fi
+
+  end_time=$(date +%s.%3N)
+  elapsed=$(echo "scale=3; $end_time - $start_time" | bc)
+  echo "Completed Deploying Hub Region A. ($elapsed)"
+
+  export ASB_SCRIPT_STEP=deployAks
+  # Save environment variables
+  ./saveenv.sh -y
+
+  # Invoke Next Step In Setup
+  $ASB_SCRIPT_STEP
+}
+
+function deployAks()
+{
+  start_time=$(date +%s.%3N)
+  echo "Deploying AKS..."
+
+  # Validate that you are using the correct vnet for cluster deployment
+  echo $ASB_SPOKE_VNET_ID
+  echo $ASB_ORG_APP_ID_NAME
+
+  # Set cluster location by choosing the closest pair - not all regions support ASB.
+  # Note: Cluster location must be the same as spoke location
+  export ASB_CLUSTER_LOCATION=${ASB_SPOKE_LOCATION}
+  export ASB_CLUSTER_GEO_LOCATION=westus
```

Comment:
This should probably also be user input since the recommendation is to use a regional pair. As is, all deployments would end with westus as the geo location.
A possible option to let the user know they can get a list of pairs here, <https://learn.microsoft.com/en-us/azure/availability-zones/cross-region-replication-azure#azure-cross-region-replication-pairings-for-all-geographies>, when they are choosing their locations.

Reply:
added prompt for geo location. good catch!